lib/sycamore/sycamore/connectors/common.py

```diff
@@ -29,25 +32,46 @@ def generate_random_string(length=8):
     return "".join(random.choice(characters) for _ in range(length))
 
 
-def filter_doc(obj, include):
-    return {k: v for k, v in obj.__dict__.items() if k in include}
+def filter_doc(doc: Document, include):
+    return {k: v for k, v in doc.items() if k in include}
 
 
 def check_dictionary_compatibility(dict1: dict[Any, Any], dict2: dict[Any, Any], ignore: list[str] = []):
     for k in dict1:
-        if ignore and any(val in k for val in ignore):
+        if not dict1.get(k) or (ignore and any(val in k for val in ignore)):
             continue
         if k not in dict2:
             return False
-        if dict1[k] != dict2[k]:
+        if dict1[k] != dict2[k] and (dict1[k] or dict2[k]):
             return False
     return True
 
 
-def compare_docs(doc1, doc2):
+def compare_docs(doc1: Document, doc2: Document):
```

Comment:
can I get a unit test or two for this fn?

Reply:
Done.